### PR TITLE
`resource/davinci_connection`: Ability to manage `defaultUserPool` connector instance

### PR DESCRIPTION
**Pull request by patrickcping**
* Bug fix: `resource/davinci_connection`: Fixed inability to import the `defaultUserPool` User Pool connector to Terraform state.
* Enhancement: `resource/davinci_connection`: Added ability to configure complex properties as JSON format.
* `resource/davinci_connection`: Moved connector reference to separate guide.
* bump `github.com/samir-gandhi/davinci-client-go` 0.1.0 => 0.2.0

### CODE REVIEW

**Comment by patrickcping** [comment]
### Testing

```shell
TF_ACC=1 go test -v -timeout 3000s -run ^TestAccResourceConnection_ github.com/pingidentity/terraform-provider-davinci/internal/service/davinci
```

### Results

```
=== RUN   TestAccResourceConnection_RemovalDrift
=== PAUSE TestAccResourceConnection_RemovalDrift
=== RUN   TestAccResourceConnection_Full_Clean
=== PAUSE TestAccResourceConnection_Full_Clean
=== RUN   TestAccResourceConnection_Full_WithBootstrap
=== PAUSE TestAccResourceConnection_Full_WithBootstrap
=== RUN   TestAccResourceConnection_Properties
=== PAUSE TestAccResourceConnection_Properties
=== RUN   TestAccResourceConnection_ComplexProperties
=== PAUSE TestAccResourceConnection_ComplexProperties
=== RUN   TestAccResourceConnection_BadParameters
=== PAUSE TestAccResourceConnection_BadParameters
=== CONT  TestAccResourceConnection_RemovalDrift
=== CONT  TestAccResourceConnection_Properties
=== CONT  TestAccResourceConnection_BadParameters
=== CONT  TestAccResourceConnection_ComplexProperties
=== CONT  TestAccResourceConnection_Full_WithBootstrap
=== CONT  TestAccResourceConnection_Full_Clean
=== NAME  TestAccResourceConnection_RemovalDrift
    resource_connection_test.go:32: Skipping step 3/4 due to SkipFunc
    resource_connection_test.go:32: Skipping step 4/4 due to SkipFunc
--- PASS: TestAccResourceConnection_RemovalDrift (75.60s)
--- PASS: TestAccResourceConnection_BadParameters (105.66s)
--- PASS: TestAccResourceConnection_ComplexProperties (141.08s)
--- PASS: TestAccResourceConnection_Properties (164.85s)
--- PASS: TestAccResourceConnection_Full_Clean (199.97s)
--- PASS: TestAccResourceConnection_Full_WithBootstrap (207.29s)
PASS
ok      github.com/pingidentity/terraform-provider-davinci/internal/service/davinci     208.105s
```

File: docs/resources/connection.md [hunk 1]
```diff
@@ -62,11 +62,11 @@ resource "davinci_connection" "my_awesome_crowdstrike_connection" {
 Required:
 
 - `name` (String) The name of the property.
-- `value` (String, Sensitive) The value of the property as string. If the property is an array, use a comma separated string.
+- `value` (String, Sensitive) The value of the property as string.  Use in conjunction with `type` to cast the value to the correct type.  For example, a number value should be entered as a string and `type` set to `number`.  JSON in string form should be used for complex types.
```
Comment (samir-gandhi) [inline docs/resources/connection.md:65]: I like this

**Comment by patrickcping** [comment]
Depends on https://github.com/samir-gandhi/davinci-client-go/pull/4

**Comment by patrickcping** [comment]
### Testing

```shell
TF_ACC=1 go test -v -timeout 3000s -run ^TestAccResourceConnection_ github.com/pingidentity/terraform-provider-davinci/internal/service/davinci
```

### Results

```
=== RUN   TestAccResourceConnection_RemovalDrift
=== PAUSE TestAccResourceConnection_RemovalDrift
=== RUN   TestAccResourceConnection_Full_Clean
=== PAUSE TestAccResourceConnection_Full_Clean
=== RUN   TestAccResourceConnection_Full_WithBootstrap
=== PAUSE TestAccResourceConnection_Full_WithBootstrap
=== RUN   TestAccResourceConnection_Properties
=== PAUSE TestAccResourceConnection_Properties
=== RUN   TestAccResourceConnection_ComplexProperties
=== PAUSE TestAccResourceConnection_ComplexProperties
=== RUN   TestAccResourceConnection_BadParameters
=== PAUSE TestAccResourceConnection_BadParameters
=== CONT  TestAccResourceConnection_RemovalDrift
=== CONT  TestAccResourceConnection_Properties
=== CONT  TestAccResourceConnection_BadParameters
=== CONT  TestAccResourceConnection_Full_WithBootstrap
=== CONT  TestAccResourceConnection_Full_Clean
=== CONT  TestAccResourceConnection_ComplexProperties
=== NAME  TestAccResourceConnection_RemovalDrift
    resource_connection_test.go:32: Skipping step 3/4 due to SkipFunc
    resource_connection_test.go:32: Skipping step 4/4 due to SkipFunc
--- PASS: TestAccResourceConnection_RemovalDrift (80.75s)
--- PASS: TestAccResourceConnection_BadParameters (108.24s)
--- PASS: TestAccResourceConnection_Properties (164.94s)
--- PASS: TestAccResourceConnection_Full_Clean (214.86s)
--- PASS: TestAccResourceConnection_Full_WithBootstrap (220.00s)
--- PASS: TestAccResourceConnection_ComplexProperties (325.69s)
PASS
ok      github.com/pingidentity/terraform-provider-davinci/internal/service/davinci     326.594s
```

**Pull request merged**